### ISSUE TITLE
Fine tuning TabICLv2 Regression

### ISSUE THREAD

**Issue by giovp** [soda-inria/tabicl#86]
Hi,

Thank you for building TabICL! I have been using for regression predictions with biological data and it is very robust. I was wondering whether there are specific strategies that you have found to work well for fine tuning tabular foundation model. I have found few papers on the topic (e.g. [this one](https://arxiv.org/pdf/2601.09654)) but was wondering if you could share some insights specifically for TabICLv2.

Thank you!

**Comment by jonaslandsgesell** [comment]
Hello Jingang

I am currently drafting a paper which outlines why it is important to enable fine-tuning for tabular foundation models

https://arxiv.org/abs/2603.08206

One main takeaway is, that the used loss function directly implies a certain inductive bias:
Even though proper scoring rules are all minimized by the true distribution in expectation, they induce different optimization trajectories and finite-sample solutions. Therefore, a pretrained model implicitly encodes a preference over distributions aligned with its training loss. Fine-tuning is not merely an optimization step, but a mechanism to redefine the decision problem the model is solving.

We also see fine-tuning impact in the ScoringBench benchmark which we started building

http://scoringbench.bolt.host/ [expect changes here since we are still styling the website]

(source code here: https://github.com/jonaslandsgesell/ScoringBench) 

So far, I was talking to David and we would love to stay in touch with you on how to measure the impacts.

Kind regards 
Jonas

**Comment by JingangQu** [comment]
Hi Jonas,

Thanks for your work and sharing your insights in fine-tuning tabular foundation models. I have not considered the role of the loss function in fine-tuning previously.

I'll read the paper and would like to stay in touch as well.

Best,
Jingang

**Comment by JingangQu** [comment]
We are trying to implment fine-tuning at this PR https://github.com/soda-inria/tabicl/pull/101

**Comment by JingangQu** [comment]
Hi, we just supported fine-tuning. Please give it a try and provide feedback thanks!

**Comment by jonaslandsgesell** [comment]
Jingang, I will run out benchmark as soon as possible

**Comment by jonaslandsgesell** [comment]
Congratulations Jingang. You shipped a nice improvement!

I ran scoringbench https://github.com/[jonaslandsgesell/ScoringBench](https://github.com/jonaslandsgesell/ScoringBench). It looks great. We will make sure to add the FinetuneTabICL results to our preprint https://arxiv.org/abs/2603.29928 and the live version of the leaderboard https://scoringbench.bolt.host/.

<img width="1505" height="832" alt="Image" src="https://github.com/user-attachments/assets/530415d8-c6bb-4428-89b8-15f2cab83aaf" /> (CD Diagram based on ScoringBench Autorank Output for R^2 )

<img width="1505" height="832" alt="Image" src="https://github.com/user-attachments/assets/cc014ea4-ff52-44b6-9e39-1b87e2c57843" />(CD Diagram based on ScoringBench Autorank Output for CRPS)

<img width="1479" height="832" alt="Image" src="https://github.com/user-attachments/assets/0e01050f-b43c-48cf-b7b9-46ee5db663b9" />(CD Diagram based on ScoringBench Autorank Output for Log Score - expected since you finetune with CRPS (integral over pinball losses) )

**Comment by JingangQu** [comment]
Cool! Thanks for your great job and validate my implementation. Great thanks!

**Comment by jonaslandsgesell** [comment]
I updated the live version of the benchmark just now (with a little bit more runs, so numbers shifted a little bit)
https://scoringbench.bolt.host/ looks great! 
Also finetuning is pretty fast.

<img width="708" height="560" alt="Image" src="https://github.com/user-attachments/assets/b76b9397-adce-4acf-97c6-1c64ae8f18ee" />
